precompiles/erc20/events_test.go

```diff
@@ -0,0 +1,73 @@
+package erc20_test
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	cmn "github.com/evmos/evmos/v15/precompiles/common"
+	erc20precompile "github.com/evmos/evmos/v15/precompiles/erc20"
+	utiltx "github.com/evmos/evmos/v15/testutil/tx"
+)
+
+//nolint:dupl // this is not a duplicate of the approval events test
+func (s *PrecompileTestSuite) TestEmitTransferEvent() {
+	s.SetupTest()
+
+	from := utiltx.GenerateAddress()
```

Comment:
You could use the keyring here instead right ?

Reply:
possibly, it's not really necessary though 🤷 we are not sending anything here, just emitting the event to the DB so and then cross-checking the address.